### PR TITLE
feat(scene composer): adding data-testid to improve e2e test

### PR DESCRIPTION
**Pull request by mumanity**
## Overview
Setting up `data-testid` handling to remove brittle logic from e2e test in scene composer. 

### Scene Composer
For `scene-composer` package changes specifically, you can preview the component in the published storybook artifact. To do this, wait for the `Publish Storybook` action to complete below.

- Click on the workflow details
- Select the Summary item on the left
- Download the zip file

To run the storybook build locally, you need a local static web server:

```
npm install -g httpserver
cd <Extracted Zip Directory>
httpserver
```

Then open the website http://localhost:8080 to run the doc site.

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: packages/scene-composer/src/components/panels/SceneHierarchyPanel/components/SceneHierarchyTree/SceneHierarchyTreeItem.tsx [hunk 1]
```diff
@@ -94,7 +94,14 @@ const SceneHierarchyTreeItem: FC<SceneHierarchyTreeItemProps> = ({
   return (
     <EnhancedTreeItem
       key={key}
-      labelNode={<SceneNodeLabel objectRef={key} labelText={labelText} componentTypes={componentTypes} />}
+      labelNode={
+        <SceneNodeLabel
+          dataTestid={node?.components[0].ref}
```
Comment (mitchlee-amzn) [inline packages/scene-composer/src/components/panels/SceneHierarchyPanel/components/SceneHierarchyTree/SceneHierarchyTreeItem.tsx:99]: Why the first component ref and not the node ref itself?

File: packages/scene-composer/src/components/panels/__tests__/__snapshots__/SceneNodeInspectorPanel.spec.tsx.snap [hunk 2]
```diff
@@ -160,7 +160,7 @@ exports[`SceneNodeInspectorPanel returns expected elements. SceneNode panel cont
           >
             <div
               data-mocked="Input"
-              data-test-id="node-ref"
+              data-testid="ip-undefined"
```
Comment (mitchlee-amzn) [inline packages/scene-composer/src/components/panels/__tests__/__snapshots__/SceneNodeInspectorPanel.spec.tsx.snap:163]: bad snapshot, should make sure the undefined is gone.

File: packages/scene-composer/src/components/three-fiber/EntityGroup/index.tsx [hunk 3]
```diff
@@ -155,7 +155,7 @@ const EntityGroup = ({ node }: IEntityGroupProps): JSX.Element => {
         onClick={onClick}
         onPointerEnter={onPointerEnter}
         onPointerLeave={onPointerLeave}
-        userData={{ nodeRef, componentTypes }}
+        userData={{ nodeRef, componentTypes, dataTestId: object3dRef?.current?.uuid }}
```
Comment (mitchlee-amzn) [inline packages/scene-composer/src/components/three-fiber/EntityGroup/index.tsx:158]: I'd just call this `testId`, the `data` part is the html standard, this isn't html.

**Review by mitchlee-amzn — APPROVED** [review]
Minor comments.

File: packages/scene-composer/src/components/panels/SceneHierarchyPanel/components/SceneHierarchyTree/SceneHierarchyTreeItem.tsx (hunk 1 above)
Comment (mumanity) [reply packages/scene-composer/src/components/panels/SceneHierarchyPanel/components/SceneHierarchyTree/SceneHierarchyTreeItem.tsx:99]: The `component` ref is the Three.js object ref. The `node` ref has no association to the Three.js object. Using `component` in order to validate that it is the correct 3D object being represented in the Inspector Panel.

File: packages/scene-composer/src/components/three-fiber/EntityGroup/index.tsx (hunk 3 above)
Comment (mumanity) [reply packages/scene-composer/src/components/three-fiber/EntityGroup/index.tsx:158]: Following standard here for all of app-kit. Using a `data` prefix bypasses the need to extend component types & lets testing remain unintrusive.

File: packages/scene-composer/src/components/panels/__tests__/__snapshots__/SceneNodeInspectorPanel.spec.tsx.snap (hunk 2 above)
Comment (mumanity) [reply packages/scene-composer/src/components/panels/__tests__/__snapshots__/SceneNodeInspectorPanel.spec.tsx.snap:163]: https://github.com/awslabs/iot-app-kit/pull/2579

**Pull request merged**